### ISSUE TITLE
java.sql.SQLNonTransientConnectionException: Could not connect to mysql:{port} : Unsupported record version Unknown-0.0

### ISSUE THREAD

**Issue by s-higuma** [metabase/metabase#9732]
I upgraded from 0.31.1 to 0.32.2. Then all questions got errors. When trying to connect to the database, I saved the database settings on the administration page, and the following error occurred.
`java.sql.SQLException: Host '10.0.3.7 'is blocked because of many connection errors; unblock with' mysqladmin flush-hosts`
Following the error, I connected to MySQL and ran `FLUSH HOSTS;`.
When I tried to save the database settings again, the following error occurred.
`java.sql.SQLNonTransientConnectionException: Could not connect to mysql: 3306: Unsupported record version Unknown-0.0`
Questions are still an error.

#### Enviroments
DB: MySQL 5.7.19(Docker)
Metabase: 0.32.2
Metabase host: Docker
Metabase DB: Postgres

If there are some mistakes in English, I 'd like to apologize.

**Comment by flamber** [comment]
Hi @s-higuma 
Metabase 0.32.3 was just released, which fixes a few issues - I would suggest that you try that version.
Do you see more details in the logs of Metabase or MySQL?
You can [increase](https://stackoverflow.com/questions/20014746/why-mysql-connection-is-blocked-of-many-connection-errors) the `max_connect_errors` from the default `100`, but it's better to find the real cause.

**Comment by s-higuma** [comment]
Upgrading to 0.32.3 did not resolve the issue. Also, there were no logs in MySQL. Metabase log is below.
#### When saving DB settings
```
2019-04-09T08: 42: 01.568474052Z 04-09 17:42:01 DEBUG middleware.log :: PUT / api / database / 2 400 307 ms (1 DB calls) Jetty threads: 8/50 (4 busy, 2 idle, 0 queued) (42 total active threads)
2019-04-09T08: 42: 01.568539272Z {: valid false,
2019-04-09T08: 42: 01.568557944Z: dbname "java.sql.SQLNonTransientConnectionException: Could not connect to mysql: 3306: Unsupported record version Unknown-0.0",
2019-04-09T08: 42: 01.568574516Z: message "java.sql.SQLNonTransientConnectionException: Could not connect to mysql: 3306: Unsupported record version Unknown-0.0"}
```
#### When a question is displayed
```
2019-04-09T09:05:31.261723694Z 04-09 18:05:31 DEBUG middleware.log :: GET /api/user/current 200 9 ms (3 DB calls) Jetty threads: 8/50 (5 busy, 4 idle, 0 queued) (50 total active threads)
2019-04-09T09:05:31.266462196Z 04-09 18:05:31 DEBUG middleware.log :: GET /api/session/properties 200 9 ms (1 DB calls) Jetty threads: 8/50 (5 busy, 4 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:31.392657388Z 04-09 18:05:31 DEBUG middleware.log :: GET /api/database 200 12 ms (3 DB calls) Jetty threads: 8/50 (4 busy, 4 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:31.603617009Z 04-09 18:05:31 DEBUG middleware.log :: GET /api/setting 200 1 ms (0 DB calls) Jetty threads: 8/50 (5 busy, 4 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:32.284837726Z 04-09 18:05:32 DEBUG middleware.log :: GET /api/database 200 803 ms (6 DB calls) Jetty threads: 8/50 (4 busy, 4 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:32.450239096Z 04-09 18:05:32 DEBUG middleware.log :: GET /api/card/11 200 20 ms (6 DB calls) Jetty threads: 8/50 (4 busy, 4 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:32.545990202Z 04-09 18:05:32 DEBUG middleware.log :: GET /api/collection/3 200 19 ms (3 DB calls) Jetty threads: 8/50 (4 busy, 4 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:32.773908375Z 04-09 18:05:32 DEBUG middleware.log :: GET /api/alert/question/11 200 4 ms (1 DB calls) Jetty threads: 8/50 (4 busy, 4 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:32.902656328Z 04-09 18:05:32 DEBUG middleware.log :: GET /api/table/21/query_metadata 200 43 ms (10 DB calls) Jetty threads: 8/50 (4 busy, 4 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:33.442529374Z 04-09 18:05:33 DEBUG middleware.log :: GET /api/table/30/query_metadata 200 21 ms (8 DB calls) Jetty threads: 8/50 (5 busy, 4 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:33.497590536Z 04-09 18:05:33 DEBUG async.util :: Running metabase.async.semaphore_channel$do_f_with_permit$f__40027@1e4f584d on separate thread...
2019-04-09T09:05:33.498527460Z 04-09 18:05:33 DEBUG async.api-response :: starting streaming response
2019-04-09T09:05:33.506057644Z 04-09 18:05:33 DEBUG middleware.log :: GET /api/table/16/query_metadata 200 52 ms (8 DB calls) Jetty threads: 8/50 (7 busy, 1 idle, 0 queued) (51 total active threads)
2019-04-09T09:05:33.513843116Z 04-09 18:05:33 DEBUG middleware.log :: GET /api/table/7/query_metadata 200 73 ms (10 DB calls) Jetty threads: 8/50 (7 busy, 1 idle, 0 queued) (52 total active threads)
2019-04-09T09:05:33.536400138Z 04-09 18:05:33 DEBUG middleware.log :: GET /api/table/18/query_metadata 200 78 ms (10 DB calls) Jetty threads: 8/50 (7 busy, 1 idle, 0 queued) (54 total active threads)
2019-04-09T09:05:33.549675313Z 04-09 18:05:33 DEBUG middleware.log :: GET /api/table/15/query_metadata 200 79 ms (10 DB calls) Jetty threads: 8/50 (6 busy, 1 idle, 0 queued) (54 total active threads)
2019-04-09T09:05:34.312310440Z 04-09 18:05:34 DEBUG middleware.log :: GET /api/table/21/fks 200 12 ms (5 DB calls) Jetty threads: 8/50 (4 busy, 3 idle, 0 queued) (56 total active threads)
2019-04-09T09:05:34.502826215Z 04-09 18:05:34 DEBUG async.api-response :: 準備中...
2019-04-09T09:05:35.077878705Z 04-09 18:05:35 ERROR sql-jdbc.execute :: タイムゾーンの設定に失敗しました： 
2019-04-09T09:05:35.077931061Z  SQLException:
2019-04-09T09:05:35.077948677Z  Message: Connections could not be acquired from the underlying database!
2019-04-09T09:05:35.077964433Z  SQLState: null
2019-04-09T09:05:35.077979589Z  Error Code: 0
2019-04-09T09:05:35.077994829Z 
2019-04-09T09:05:35.078087685Z 04-09 18:05:35 ERROR sql-jdbc.execute :: タイムゾーンの設定に失敗しました： 
2019-04-09T09:05:35.078106861Z  SQLException:
2019-04-09T09:05:35.078122221Z  Message: Connections could not be acquired from the underlying database!
2019-04-09T09:05:35.078162097Z  SQLState: null
2019-04-09T09:05:35.078177889Z  Error Code: 0
2019-04-09T09:05:35.078205345Z 
2019-04-09T09:05:35.081720099Z SQLException:
2019-04-09T09:05:35.081773307Z  Message: Connections could not be acquired from the underlying database!
2019-04-09T09:05:35.081790731Z  SQLState: null
2019-04-09T09:05:35.081806259Z  Error Code: 0
2019-04-09T09:05:35.082020915Z 04-09 18:05:35 ERROR sql-jdbc.execute :: nil
2019-04-09T09:05:35.082757836Z SQLException:
2019-04-09T09:05:35.082800364Z  Message: Connections could not be acquired from the underlying database!
2019-04-09T09:05:35.082817188Z  SQLState: null
2019-04-09T09:05:35.082832560Z  Error Code: 0
2019-04-09T09:05:35.083465188Z 04-09 18:05:35 ERROR sql-jdbc.execute :: nil
2019-04-09T09:05:35.489988686Z 04-09 18:05:35 WARN metabase.query-processor :: Query failure {:status :failed,
2019-04-09T09:05:35.490044918Z  :class java.lang.Exception,
2019-04-09T09:05:35.490063446Z  :error "Connections could not be acquired from the underlying database!",
2019-04-09T09:05:35.490080174Z  :stacktrace
2019-04-09T09:05:35.490095558Z  ("--> driver.sql_jdbc.execute$do_with_try_catch.invokeStatic(execute.clj:222)"
2019-04-09T09:05:35.490111758Z   "driver.sql_jdbc.execute$do_with_try_catch.invoke(execute.clj:213)"
2019-04-09T09:05:35.490127730Z   "driver.sql_jdbc.execute$execute_query.invokeStatic(execute.clj:283)"
2019-04-09T09:05:35.490144230Z   "driver.sql_jdbc.execute$execute_query.invoke(execute.clj:277)"
2019-04-09T09:05:35.490159998Z   "driver.sql_jdbc$fn__62464.invokeStatic(sql_jdbc.clj:48)"
2019-04-09T09:05:35.490175586Z   "driver.sql_jdbc$fn__62464.invoke(sql_jdbc.clj:47)"
2019-04-09T09:05:35.490190994Z   "query_processor$fn__39544$execute_query__39549$fn__39550.invoke(query_processor.clj:67)"
2019-04-09T09:05:35.490221630Z   "query_processor$fn__39544$execute_query__39549.invoke(query_processor.clj:61)"
2019-04-09T09:05:35.490237674Z   "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__33901.invoke(mbql_to_native.clj:36)"
2019-04-09T09:05:35.490253250Z   "query_processor.middleware.annotate$result_rows_maps__GT_vectors$fn__32622.invoke(annotate.clj:368)"
2019-04-09T09:05:35.490269234Z   "query_processor.middleware.annotate$add_column_info$fn__32527.invoke(annotate.clj:303)"
2019-04-09T09:05:35.490285938Z   "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__33364.invoke(cumulative_aggregations.clj:57)"
2019-04-09T09:05:35.490301706Z   "query_processor.middleware.limit$limit$fn__33857.invoke(limit.clj:19)"
2019-04-09T09:05:35.490318986Z   "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__39411.invoke(results_metadata.clj:86)"
2019-04-09T09:05:35.490353378Z   "query_processor.middleware.format_rows$format_rows$fn__33845.invoke(format_rows.clj:26)"
2019-04-09T09:05:35.490369914Z   "query_processor.middleware.add_dimension_projections$add_remapping$fn__31679.invoke(add_dimension_projections.clj:236)"
2019-04-09T09:05:35.490386282Z   "query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__32073.invoke(add_row_count_and_status.clj:15)"
2019-04-09T09:05:35.490402350Z   "query_processor.middleware.driver_specific$process_query_in_context$fn__33606.invoke(driver_specific.clj:12)"
2019-04-09T09:05:35.490418106Z   "query_processor.middleware.resolve_driver$resolve_driver$fn__36377.invoke(resolve_driver.clj:15)"
2019-04-09T09:05:35.490433742Z   "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__32802$fn__32803.invoke(bind_effective_timezone.clj:9)"
2019-04-09T09:05:35.490449810Z   "util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
2019-04-09T09:05:35.490465218Z   "util.date$call_with_effective_timezone.invoke(date.clj:77)"
2019-04-09T09:05:35.490480710Z   "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__32802.invoke(bind_effective_timezone.clj:8)"
2019-04-09T09:05:35.490496850Z   "query_processor.middleware.store$initialize_store$fn__39436$fn__39437.invoke(store.clj:11)"
2019-04-09T09:05:35.490512738Z   "query_processor.store$do_with_new_store.invokeStatic(store.clj:35)"
2019-04-09T09:05:35.490528254Z   "query_processor.store$do_with_new_store.invoke(store.clj:31)"
2019-04-09T09:05:35.490543758Z   "query_processor.middleware.store$initialize_store$fn__39436.invoke(store.clj:10)"
2019-04-09T09:05:35.490559082Z   "query_processor.middleware.cache$maybe_return_cached_results$fn__33224.invoke(cache.clj:127)"
2019-04-09T09:05:35.490574622Z   "query_processor.middleware.catch_exceptions$catch_exceptions$fn__33272.invoke(catch_exceptions.clj:63)"
2019-04-09T09:05:35.490590138Z   "query_processor$process_query.invokeStatic(query_processor.clj:232)"
2019-04-09T09:05:35.490605534Z   "query_processor$process_query.invoke(query_processor.clj:228)"
2019-04-09T09:05:35.490620834Z   "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:344)"
2019-04-09T09:05:35.490636266Z   "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:338)"
2019-04-09T09:05:35.490651602Z   "query_processor$fn__39625$process_query_and_save_execution_BANG___39630$fn__39631.invoke(query_processor.clj:371)"
2019-04-09T09:05:35.490667250Z   "query_processor$fn__39625$process_query_and_save_execution_BANG___39630.invoke(query_processor.clj:357)"
2019-04-09T09:05:35.490683090Z   "async.semaphore_channel$do_f_with_permit$f__40027.invoke(semaphore_channel.clj:81)"
2019-04-09T09:05:35.490698762Z   "async.util$do_on_separate_thread$fn__39862.invoke(util.clj:53)"),
2019-04-09T09:05:35.490714218Z  :query
2019-04-09T09:05:35.490728990Z  {:type :query,
2019-04-09T09:05:35.490743894Z   :query {:source-table 21, :filter [:time-interval [:field-id 107] :current :month], :aggregation [[:count]]},
2019-04-09T09:05:35.490768446Z   :constraints {:max-results 10000, :max-results-bare-rows 2000},
2019-04-09T09:05:35.490784226Z   :parameters [],
2019-04-09T09:05:35.490799250Z   :middleware nil,
2019-04-09T09:05:35.490815774Z   :cache-ttl nil,
2019-04-09T09:05:35.490830546Z   :info
2019-04-09T09:05:35.490845510Z   {:executed-by 1,
2019-04-09T09:05:35.490860414Z    :context :question,
2019-04-09T09:05:35.490875126Z    :card-id 11,
2019-04-09T09:05:35.490889910Z    :dashboard-id nil,
2019-04-09T09:05:35.490905186Z    :query-hash [-94, 78, 15, -101, -12, 7, -10, 101, -6, -104, 35, -36, 77, -89, 16, -84, 29, -90, 112, -61, 7, 68, 7, 61, -2, -103, 122, -123, 21, 103, -62, -15],
2019-04-09T09:05:35.490924794Z    :query-type "MBQL"}},
2019-04-09T09:05:35.490940718Z  :preprocessed
2019-04-09T09:05:35.490956078Z  {:constraints {:max-results 10000, :max-results-bare-rows 2000},
2019-04-09T09:05:35.490972554Z   :settings {:report-timezone "Asia/Tokyo"},
2019-04-09T09:05:35.490987938Z   :type :query,
2019-04-09T09:05:35.491002602Z   :info
2019-04-09T09:05:35.491017398Z   {:executed-by 1,
2019-04-09T09:05:35.491032170Z    :context :question,
2019-04-09T09:05:35.491046834Z    :card-id 11,
2019-04-09T09:05:35.491061378Z    :query-hash [-94, 78, 15, -101, -12, 7, -10, 101, -6, -104, 35, -36, 77, -89, 16, -84, 29, -90, 112, -61, 7, 68, 7, 61, -2, -103, 122, -123, 21, 103, -62, -15],
2019-04-09T09:05:35.491076918Z    :query-type "MBQL"},
2019-04-09T09:05:35.491091918Z   :database 2,
2019-04-09T09:05:35.491106690Z   :driver :mysql,
2019-04-09T09:05:35.491121750Z   :query {:source-table 21, :filter [:= [:datetime-field [:field-id 107] :month] [:relative-datetime 0 :month]], :aggregation [[:count]]}},
2019-04-09T09:05:35.491137254Z  :native
2019-04-09T09:05:35.491151858Z  {:query
2019-04-09T09:05:35.491167158Z   "SELECT ~ (Omitted because the query content is written)
2019-04-09T09:05:35.491183574Z   :params nil}}
2019-04-09T09:05:35.491206626Z 
2019-04-09T09:05:35.504691177Z 04-09 18:05:35 DEBUG async.api-response :: 準備中...
2019-04-09T09:05:35.516439926Z 04-09 18:05:35 DEBUG async.semaphore-channel :: f finished, permit will be returned
2019-04-09T09:05:35.518127955Z 04-09 18:05:35 DEBUG async.api-response :: Async response finished, closing channels.
2019-04-09T09:05:35.521581473Z 04-09 18:05:35 DEBUG middleware.log :: POST /api/card/11/query 200 [ASYNC: completed] 2 s (19 DB calls) Jetty threads: 8/50 (3 busy, 3 idle, 0 queued) (56 total active threads)
```

After creating and running a new Metabase and a new Postgresql container, Metabase successfully communicated with MySQL.
I tried importing Postgresql's dump file into a new Postgresql, but a similar error occurred.

**Comment by ozasuke** [comment]
+1

**Comment by camsaul** [comment]
Related: #9483 and #9629

**Comment by camsaul** [comment]
@s-higuma can you give me the exact Docker image you're using for your MySQL DB. I will try this out and see if I can repro it. If I can hopefully I can fix it

**Comment by camsaul** [comment]
Did a search for 

```
Unsupported record version Unknown-0.0
```

and it sounds like it's another MariaDB JDBC driver x SSL issue

[This page](https://mariadb.com/kb/en/library/using-tls-ssl-with-mariadb-java-connector/) has tons of information on the issue.

**Comment by camsaul** [comment]
During TLS/SSL negotiation, the client normally indicates the maximum level supported, but before 10.2, YaSSL doesn't recognize the TLSv1.2 flag and then doesn't fallback to theTLSv1.1 protocol.

> If trying to connect with TLSv1.2 with a server before 10.2 using YaSSL, the connection will fail with the following exception :
> 
> with MariaDB server : "Caused by: javax.net.ssl.SSLException: Unsupported record version Unknown-0.0"
> with MySQL server : "Caused by: javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake"

**Comment by camsaul** [comment]
This is actually a duplicate of #9629

**Comment by s-higuma** [comment]
@camsaul Sorry for late reply. Thank you for the information. Provides information about mysql images. Please refer to the following. Is this enough information?
```
root@{container_id}:/# mysql --version
mysql: [Warning] World-writable config file '/etc/mysql/conf.d/custom.cnf' is ignored.
mysql  Ver 14.14 Distrib 5.7.19, for Linux (x86_64) using  EditLine wrapper
```
```
[~] # docker inspect {image_id}
[
    {
        "Id": "sha256:c73c7527c03a13729a6bba88a3251c95629e8036c0ccf17dadefb4c5f077a315",
        "RepoTags": [
            "mysql:latest"
        ],
        "RepoDigests": [],
        "Parent": "",
        "Comment": "",
        "Created": "2017-07-26T07:06:10.743040414Z",
        "Container": "9c87a2f5a490532ea8335271fa7a9b4b8bcd4a4241cf2d54621829f39da41b23",
        "ContainerConfig": {
            "Hostname": "e24d0b5cccc1",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "3306/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "GOSU_VERSION=1.7",
                "MYSQL_MAJOR=5.7",
                "MYSQL_VERSION=5.7.19-1debian8"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "#(nop) ",
                "CMD [\"mysqld\"]"
            ],
            "ArgsEscaped": true,
            "Image": "sha256:de06bc71fc431f5c122026d63c39b1d48a80fcbff79449633979389d626e0c57",
            "Volumes": {
                "/var/lib/mysql": {}
            },
            "WorkingDir": "",
            "Entrypoint": [
                "docker-entrypoint.sh"
            ],
            "OnBuild": [],
            "Labels": {}
        },
        "DockerVersion": "17.03.2-ce",
        "Author": "",
        "Config": {
            "Hostname": "e24d0b5cccc1",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "3306/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "GOSU_VERSION=1.7",
                "MYSQL_MAJOR=5.7",
                "MYSQL_VERSION=5.7.19-1debian8"
            ],
            "Cmd": [
                "mysqld"
            ],
            "ArgsEscaped": true,
            "Image": "sha256:de06bc71fc431f5c122026d63c39b1d48a80fcbff79449633979389d626e0c57",
            "Volumes": {
                "/var/lib/mysql": {}
            },
            "WorkingDir": "",
            "Entrypoint": [
                "docker-entrypoint.sh"
            ],
            "OnBuild": [],
            "Labels": {}
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 412339426,
        "VirtualSize": 412339426,
        "GraphDriver": {
            "Name": "devicemapper",
            "Data": {
                "DeviceId": "122",
                "DeviceName": "docker-252:0-3407887-7496afa594c7bf828327738f45135cf8f1e6f9b5cd26fe14ac4c4ae0028102b1",
                "DeviceSize": "10737418240"
            }
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:2c40c66f7667aefbb18f7070cf52fae7abbe9b66e49b4e1fd740544e7ceaebdc",
                "sha256:e4c3daa6e001893636bda0ed59bba8bd2c5024f453f13273c68c9378f0628841",
                "sha256:b56bff4e0584f65a4a3859ce49cca678d8f83505d878091a2041a212fd99002a",
                "sha256:7de0cc4a043caf98aa19b4f5e6cf38c0ecf6acc2f87294772793562512993cc7",
                "sha256:a7d38defd5b8c498716cbfcace36300d88b7e14f2034ad0d2ac3e8d6ff6e9b55",
                "sha256:19f8de075fd393e472663b6a60d250bb253b7f0ec461fd32f13c3cd345870d42",
                "sha256:0032f17d7e40fd705571666c9ac4302bf8a9ae153f3c72e4b4516e7a7602df68",
                "sha256:744a3477d6d6248cff6e2941688390683d1596e673c98e4a64e3baefa24d8ccb",
                "sha256:12f318f3f22d2963bacaedd1f05a52d112011ab90fdc4ee6524c61f4296bab9a",
                "sha256:484751db18d22a6755b8be80cdb2b4b5b65d401be4dbc0c99b92bfe5bd18d780",
                "sha256:897ad04f38a61b03843afa42b4ab80c840fccde369613a5668f7a2b516f300fa"
            ]
        },
        "Remote": false
    }
]
```

If there are some mistakes in English, I 'd like to apologize.

**Comment by eladeyal-intel** [comment]
@s-higuma if this is still a problem - see the resolution of   #9629  - try to supply the SSL version explicitly.
